root/programs/C#/Frameworks/Infrastructure/Framework/Transmission/CallController.cs

```diff
@@ -1253,7 +847,233 @@ public object Invoke(string serviceName, object parameterValue)
+                    // XML定義：キーが無い
+                }
+                else
+                {
+                    if (props[FxLiteral.TRANSMISSION_HTTP_PROP_USER_NAME] == null
+                        || (string)props[FxLiteral.TRANSMISSION_HTTP_PROP_USER_NAME] == "")
+                    {
+                        // XML定義：null or 空文字列
+                    }
+                    else
+                    {
+                        // XML定義：あり
+
+                        // WASのセキュリティ資格情報を生成する。
+                        nwcWAS = new NetworkCredential();
+                        nwcWAS.UserName = (string)props[FxLiteral.TRANSMISSION_HTTP_PROP_USER_NAME];
+                        nwcWAS.Password = (string)props[FxLiteral.TRANSMISSION_HTTP_PROP_PASSWORD];
+
+                        // 省略可能に変更した。
+                        if (props.ContainsKey(FxLiteral.TRANSMISSION_HTTP_PROP_DOMAIN))
+                        {
+                            nwcWAS.Domain = (string)props[FxLiteral.TRANSMISSION_HTTP_PROP_DOMAIN];
+                        }
+                    }
+                }
+            }
+            else
+            {
+                // ユーザ指定：あり
+
+                // WASのセキュリティ資格情報を設定する（ユーザ指定）。
+                nwcWAS = this._nwcWAS;
+            }
+
+            return nwcWAS;
+        }
+
+        /// <summary>CreateCredentials</summary>
+        /// <param name="props">Dictionary(string, string)</param>
+        /// <returns>X509Certificate</returns>
+        private X509Certificate2 CreateX509Certificate(Dictionary<string, string> props)
+        {
+            X509Certificate2 x509 = null;
+
+            // http://support.microsoft.com/kb/895971/ja
+            // http://msdn.microsoft.com/ja-jp/library/system.security.cryptography.x509certificates.x509certificate.x509certificate.aspx
+
+            if (!props.ContainsKey(FxLiteral.TRANSMISSION_HTTP_PROP_X509CERTIFICATE_FILE))// Dic化でnullチェック変更
+            {
+                // XML定義：キーが無い
+            }
+            else
+            {
+                if (props[FxLiteral.TRANSMISSION_HTTP_PROP_X509CERTIFICATE_FILE] == null
+                    || (string)props[FxLiteral.TRANSMISSION_HTTP_PROP_X509CERTIFICATE_FILE] == "")
+                {
+                    // XML定義：null or 空文字列
+                }
+                else
+                {
+                    if (!props.ContainsKey(FxLiteral.TRANSMISSION_HTTP_PROP_X509CERTIFICATE_PASSWORD))// Dic化でnullチェック変更
+                    {
+                        // XML定義：キーが無い
+
+                        // クライアント証明書のファイルパス
+                        x509 = new X509Certificate2((string)props[FxLiteral.TRANSMISSION_HTTP_PROP_X509CERTIFICATE_FILE]);
+                    }
+                    else
+                    {
+                        if (props[FxLiteral.TRANSMISSION_HTTP_PROP_X509CERTIFICATE_PASSWORD] == null
+                            || (string)props[FxLiteral.TRANSMISSION_HTTP_PROP_X509CERTIFICATE_PASSWORD] == "")
```

Comment:
@daisukenishino 

Similar to [this](https://github.com/OpenTouryoProject/OpenTouryo/pull/253/files#r134915067).